Content/Items/Gear/Weapons/Battleaxe/GuardianAngel.cs

```diff
@@ -0,0 +1,154 @@
+﻿using System.Collections.Generic;
+using PathOfTerraria.Content.Projectiles.Melee;
+using PathOfTerraria.Core.Systems;
+using PathOfTerraria.Core.Systems.Affixes;
+using PathOfTerraria.Core.Systems.Affixes.ItemTypes.WeaponAffixes;
+using ReLogic.Content;
+using Terraria.Localization;
+
+namespace PathOfTerraria.Content.Items.Gear.Weapons.Battleaxe;
+
+internal class GuardianAngel : SteelBattleaxe
+{
+	public override float DropChance => 1f;
+	public override bool IsUnique => true;
+	public override string AltUseDescription => Language.GetTextValue("Mods.PathOfTerraria.Items.GuardianAngel.AltUseDescription");
+	public override string Description => Language.GetTextValue("Mods.PathOfTerraria.Items.GuardianAngel.Description");
+	public override int MinDropItemLevel => 25;
+
+	public override void Defaults()
+	{
+		base.Defaults();
+		Item.width = 94;
+		Item.height = 108;
+	}
+	
+	public override bool AltFunctionUse(Player player)
+	{
+		AltUsePlayer modPlayer = player.GetModPlayer<AltUsePlayer>();
+
+		if (!modPlayer.AltFunctionAvailable)
+		{
+			return false;
+		}
+
+		foreach (NPC npc in Main.ActiveNPCs)
+		{
+			if (npc.CanBeChasedBy() && npc.DistanceSQ(player.Center) < 400 * 400)
+			{
+				npc.GetGlobalNPC<AngelRingNPC>().ApplyRing(player.whoAmI);
+			}
+		}
+
+		modPlayer.SetAltCooldown(180, 0);
+		return true;
+	}
+	
+	public override void OnHitNPC(Player player, NPC target, NPC.HitInfo hit, int damageDone)
+	{
+		target.GetGlobalNPC<AngelRingNPC>().ApplyRing(player.whoAmI);
+	}
+
+	public override List<ItemAffix> GenerateAffixes()
+	{
+		var addedDamageAffix = (ItemAffix)Affix.CreateAffix<PassiveAffixes.AddedDamageAffix>();
+		addedDamageAffix.MinValue = 1;
+		addedDamageAffix.MaxValue = 4;
+		
+		var increasedDamageAffix = (ItemAffix)Affix.CreateAffix<ModifyHitAffixes.BaseKnockbackItemAffix>();
+		increasedDamageAffix.Value = 0.1f;
```

Comment:
Think we want to use min and max value equaling the same for rerolling if we want set values.